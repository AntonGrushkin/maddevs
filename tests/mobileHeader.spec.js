import { mount } from '@vue/test-utils';
import MobileHeader from '@/components/Header/mobile-header';

describe('MobileHeader', () => {
  let wrapper;
  let getMountParams = includesValue => {
    let mountParams = {
      stubs: ['router-link', 'headerLogo'],
      mocks: {
        $nuxt: {
          $route: {
            path: {
              includes: jest.fn(() => includesValue)
            }
          }
        }
      }
    };
    return mountParams;
  };
  const htmlElementsId = ['case-main-container', 'read-our-case-btn', 'case-first-section', 'case-main-container', 'case-header', 'header-logo-text'];
  htmlElementsId.forEach(elementId => {
    let element = document.createElement('div');
    element.setAttribute('id', elementId);
    element.getBoundingClientRect = jest.fn(() => ({ // Set custom value for getBoundingClientRect
      top: 100
    }));
    document.body.appendChild(element);
  });
  beforeEach(() => {
    wrapper = mount(MobileHeader, getMountParams(true));
  });

  // ------ IMPORTANT ----- //
  test('is a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy();
  });
  
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  // --------------------- //
});
