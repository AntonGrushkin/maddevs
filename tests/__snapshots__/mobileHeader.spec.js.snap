// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobile Header renders correctly 1`] = `
<div
  class="mobile-header"
>
  <div
    class="mobile-header__top-line"
  >
    <nuxt-link-stub
      class="mobile-header__header-logo"
      to="/"
    >
      <img
        alt="Logotype"
        src="@/assets/img/common/logo.svg"
      />
    </nuxt-link-stub>
     
    <div
      class="mobile-header__button-wrap"
    >
      <button
        class="mobile-header__toogle-btn"
      >
        <img
          alt="Open"
          class="mobile-header__open"
          src="@/assets/img/Header/mobile-header-hamburger.svg"
        />
      </button>
    </div>
  </div>
   
  <perfect-scrollbar-stub
    class="mobile-header__scrollbar custom-scrollbar container"
  >
    <div
      class="mobile-header__content-wrap"
      style="display: none;"
    >
      <nav
        class="mobile-header__header-routes_links"
      >
        <nuxt-link-stub
          exact=""
          to="/"
        >
          About
        </nuxt-link-stub>
         
        <nuxt-link-stub
          to="/services"
        >
          Services
        </nuxt-link-stub>
         
        <nuxt-link-stub
          to="/projects"
        >
          Projects
        </nuxt-link-stub>
         
        <nuxt-link-stub
          to="/careers"
        >
          Careers
        </nuxt-link-stub>
         
        <a
          href="https://blog.maddevs.io/"
          rel="noreferrer"
          target="_blank"
        >
          Blog
        </a>
      </nav>
       
      <button
        class="ui-button-modal-trigger mobile-header__modal-trigger-btn ui-button-modal-trigger--red"
      >
        
  Contact me

      </button>
       
      <div
        class="mobile-header__contacts"
      >
        <div
          class="footer-contacts"
        >
          <div
            class="footer-contacts__call-us"
          >
            <h4
              class="footer-contacts__title"
            >
              Call us:
            </h4>
             
            <ul
              class="footer-contacts__phones-list"
            >
              <li>
                <a
                  class="footer-contacts__phone-item footer-contacts__phone-item--flag"
                  href="tel:+44 20 3984 8555"
                >
                  <span
                    class="footer-contacts__icon footer-contacts__icon--uk"
                  />
                  
					+44 20 3984 8555
				
                </a>
              </li>
            </ul>
          </div>
           
          <div
            class="footer-contacts__message-us"
          >
            <h4
              class="footer-contacts__title"
            >
              Message us:
            </h4>
             
            <ul
              class="footer-contacts__messengers-list"
            >
              <li>
                <a
                  class="footer-contacts__messenger-item"
                  href="https://facebook.com/maddevsllc"
                  target="_blank"
                >
                  <span
                    class="footer-contacts__message footer-contacts__message--messenger"
                  />
                  
					Messenger
				
                </a>
              </li>
               
              <li>
                <a
                  class="footer-contacts__messenger-item"
                  href="http://wa.me/996555771481"
                  target="_blank"
                >
                  <span
                    class="footer-contacts__message footer-contacts__message--whatsapp"
                  />
                  
					Whatsapp
				
                </a>
              </li>
               
              <li>
                <a
                  class="footer-contacts__messenger-item"
                  href="https://t.me/mun_tamara/"
                  target="_blank"
                >
                  <span
                    class="footer-contacts__message footer-contacts__message--telegram"
                  />
                  
					Telegram
				
                </a>
              </li>
               
              <li>
                <a
                  class="footer-contacts__messenger-item"
                  href="https://msng.link/o/?https%3A%2F%2Fu.wechat.com%2FICWluRgJH8tu0IisMQ1eEFo=wc"
                  target="_blank"
                >
                  <span
                    class="footer-contacts__message footer-contacts__message--wechat"
                  />
                  
					WeChat
				
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
       
      <div
        class="mobile-header__social-network_links"
      >
        <div
          class="footer-social-network"
        >
          <a
            class="footer-social-network__link footer-social-network__link--github"
            href="https://www.facebook.com/maddevsllc"
            target="_blank"
          />
          <a
            class="footer-social-network__link footer-social-network__link--facebook"
            href="https://www.facebook.com/maddevsllc"
            target="_blank"
          />
          <a
            class="footer-social-network__link footer-social-network__link--instagram"
            href="https://www.instagram.com/maddevsio/"
            target="_blank"
          />
          <a
            class="footer-social-network__link footer-social-network__link--medium"
            href="https://blog.maddevs.io/"
            target="_blank"
          />
          <a
            class="footer-social-network__link footer-social-network__link--twitter"
            href="https://twitter.com/MadDevsIO"
            target="_blank"
          />
          <a
            class="footer-social-network__link footer-social-network__link--behance"
            href="https://www.behance.net/maddevs"
            target="_blank"
          />
        </div>
      </div>
    </div>
  </perfect-scrollbar-stub>
</div>
`;
